tests/test_connection_id.py

```
@@ -0,0 +1,87 @@
+import psycopg
+import unittest
+
+
+def _ensure_riffq_built():
+    subprocess.call([sys.executable, "-m", "pip", "uninstall", "-y", "riffq"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    try:
+        subprocess.check_call(["maturin", "build", "--release", "-q"])
+        wheel = next(Path("target/wheels").glob("riffq-*.whl"))
+        subprocess.check_call([sys.executable, "-m", "pip", "install", str(wheel)])
+    except Exception as exc:
+        raise RuntimeError(f"riffq build failed: {exc}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)